---
title: "Introduction to joincount"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction to joincount}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup, message = FALSE, warning=FALSE}
library(joincount)
library(spdep)
library(raster)
library(pgirmess)
library(Seurat)
library(rgdal)
library(ggplot2)
library(reshape2)
library(lattice)
library(grid)
```

v0.99.0

joincount: Quantification tool for spatial correlation between clusters in spatial transciptomics pre-processed data using join count statustic approach.  

This tool requires data produced with the 10X Genomics Visium Spatial Gene Expression platform with either standard or customized clusters. The purpose of this R package is to perform join count analysis for spatial correlation between clusters. Examples of how to run this tool are below:  

### Important to know before processing:
There are two types of orientation of the slide, here they are defined as "Vertical" and "Horizontal".  

Vertical - The distance of those points in the vertical direction is smaller than their distance in the horizontal direction.  

Horizontal - The distance of those points in the Horizontal direction is smaller than their distance in the vertical direction.  

User can check the orientation information from the image `./spatial/detected_tissue_image.jpg`  

Those functions end-up with `Ver` are designed for the vertical orientation slides.  

1. spatialDataPrep() is a function to load a 10x Genomics Visium Spatial Experiment into a Seurat object, followed by sctransform and add seurat cluster label that provided in `./analysis/clustering/graphclust` to corresponding barcode. The 
input is a directory containing the H5 file specified by filename, the image data in a subdirectory called spatial, and analysis results in a subdirectory called analysis.
```{r dataImport, warnings = FALSE}
sample <- spatialDataPrep('../inst/extdata/Example')
# The name of .h5 file must match to the name of filtered_feature_bc_matrix
```
* Note: User can skip this step if they want to use the data with customized clusters attached. The data format required by this package is the column name of customized cluster labels must be called "Cluster". It can be checked by `head(sample@meta.data)` and can be changed as need.

2. resolutionCalc() (resolutionCalcVer() for vertical orientation) is a function to calculating the resolution for rasterization. This function will return a list of resolution.
```{r, warnings = FALSE}
resolutionList <- resolutionCalc(sample)
```

3. rasterPrep() (rasterPrepVer() for vertical orientation) is a function to find the extent and set up the raster. This function will return a  rasterLayer object.
```{r, warnings = FALSE}
raster <- rasterPrep(sample)
```

4. rasterizeEachCluster() (rasterizeEachClusterVer() for vertical orientation) is a function to rasterize each cluster and return the integrated mosaic.  

User can directly jump to this step without perform resolutionCalc() and rasterPrep().  
```{r, warnings = FALSE}
mosaicIntegration <- rasterizeEachCluster(sample)
```

5. mosaicIntPlot() to visualize the integrated mosaic.
```{r, fig.dim = c(15, 15)}
mosaicIntPlot(mosaicIntegration)
```
6. joincountAnalysis() is a function that take the output of rasterizeEachCluster() as the input and return Join count analysis results. The result contain cluster pair/ observed join count/ Expected join count under spatial randomness/ Variance of observed vs expected/ Z-score.
```{r, warnings = FALSE}
joincount.result <- joincountAnalysis(mosaicIntegration)
```

